From 40ad62a697f4dd4e6214e5788c77c718d0a0165a Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Fri, 25 Nov 2016 16:00:53 -0800
Subject: [PATCH 126/173] SnapdragonCamera: Fix shutter button clicks in rapid
 succession getting ignored

Due to this check, the shutter button would be disabled until each photo
would finish processing. This results in the shutter button getting
disabled for quite a long time when ZSL isn't used, forcing the user to
wait until the previous shot is finished before requesting a new shot.

Fix the HDR check so that shutter button clicks in quick succession work.
There is code in place to handle button clicks in rapid succession, so this
is fine.

Change-Id: Ifc36a22e4ec3a1be24e5a9b40c3fc682e82605b3
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 src/com/android/camera/PhotoModule.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index b838cfb34..a7fbfa662 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -1837,7 +1837,7 @@ public class PhotoModule
 
         // We don't want user to press the button again while taking a
         // multi-second HDR photo. For longshot, no need to disable.
-        if (mCameraState != LONGSHOT) {
+        if (CameraUtil.SCENE_MODE_HDR.equals(mSceneMode)) {
             mUI.enableShutter(false);
         }
 
-- 
2.17.1

